designdocs/delegation.md

```diff
@@ -22,175 +22,129 @@ The primary motivation for delegation is to support End Users of the DSNP platfo
 
 ## Proposal
-The proposed solution is to give End Users the ability to create an on-chain `MsaId` through an authorized provider. End Users can also transparently authorize and manage their own Providers and permissions, either directly using a native token or through an explicitly authorized Provider. Additionally, we allow `MsaIds` to be directly purchased using a native token.
+The proposed solution is to give End Users the ability to create an on-chain MSA id` through an authorized provider. End Users can also transparently authorize and manage their own Providers and permissions, either directly using a native token or through an explicitly authorized Provider. Additionally, we allow MSA ids` to be directly purchased using a native token.
 
 ### API (extrinsics)
 * All names are placeholders and may be changed.
 * All extrinsics must emit an appropriate event with all parameters for the call, unless otherwise specified.
 * Errors in the extrinsics must have different, reasonably-named error enums for each type of error for ease of debugging.
-* "Owner only" means the caller must own the delegator`MsaId`.
+* "Owner only" means the caller must own the delegatorMSA id`.
 * Events are not deposited for read-only extrinsic calls.
 
-#### create_account_with_provider
-Creates a new `MsaId` on behalf of an MSA and adds the origin as the MSA's provider.
+#### create_sponsored_account_with_delegation
+Creates a new MSA on behalf of a delegator and adds the origin held MSA as its provider.
+
   * Parameters:
-      1. `payload`:
-         1. `data` - this is what the MSA owner must sign and provide to the provider beforehand.
-             * `msa_id` - the provider, of type `MsaId`
-             * `permission` a value indicating the permission to be given to the provider
-         2. `public_key` - The authorizing key used to create `signature`
-         3. `signature` - The signature of the hash of `data`
-  * Event:  `IdentityCreated`, with the `public_key`, the newly created `MsaId`, and `msa_id`
-  * Restrictions:  origin must own `msa_id` in the payload.
-
-#### add_self_as_provider(payload)
-Adds the `MsaId` in the payload as a provider, to an MSA owning `delegator_msa_id`
+      1. `add_provider_payload` - this is what the holder of delegator_key must sign and provide to the provider beforehand.
+          * `authorized_msa_id` - the provider, of type `MessageSourceId`
+          * `permission` a value indicating the permission to be given to the provider
+      2. `delegator_key` - The authorizing key used to create `proof`
+      3. `proof` - The signature of the hash of `add_provider_payload` by the delgator
+
+  * Events:
+      1. `MsaCreated`
+          * `new_msa_id` - id of the newly created MSA
+          * `key` - the `delegator_key`
+      2. `ProviderAdded`
+          * `delegator` - id of the newly created MSA
+          * `provider` - id of the MSA help by the provider
+
+#### remove_delegation_by_provider
```

Comment:
Should this also be named `revoke` to match?

Reply:
This was named `remove_delegation_by_provider` in this document to match the name of the currently existing extrinsic, however if you mean to rename the extrinsic and its references to match other nomenclature including the `revoke_msa_delegation_by_delegator` extrinsic, then yes I agree

Reply:
I modified and created an issue to rename the extrinsic:
https://github.com/LibertyDSNP/frequency/issues/369